### DISCUSSION TITLE
Farming System

### DISCUSSION THREAD

**Discussion by thebonbon** [EveronLife/EveronLife#61] [Ideas]
### Basic building / farming concept

https://user-images.githubusercontent.com/45827986/171268175-8d7490b6-7833-43d4-ab23-b60c5d1b09ec.mp4

I tried to make it as modular as possible. Here is the Inheritance of the tomato crop:

`GenericEntity -> EL_BaseBuilding -> EL_BaseCrop -> EL_PositionCrop`

**`EL_BaseBuilding`** is a base for the building system. It currently only has one enum to set the type of building (For later use)
```
enum EBuildingType
{
	CROP
};
```

(This can be expanded to furniture etc..)

**`EL_BaseCrop `** is a base for all crops. It calls the Grow method (just returns true) onFrame and sets the needed event masks.

**`EL_PositionCrop`** is a final crop type. It overrides the Grow method to move the crop up a little each frame (Y Pos). (As seen in video)

To fit this system I also added two variables to the EL_GatherAction. This is the tomato gather action:
![image](https://user-images.githubusercontent.com/45827986/171271971-42085ed8-3a3e-44a3-b83f-f831237e6a1b.png)

`Delete After Use` Deletes gathered object after use (As seen in video)
`Always Active` to support the current tree gathering (arma 3 style)

**Crop Ideas:**

`EL_StagedCrop` multi staged crop type. Grows to a new stage after x time.
`EL_SizeCrop` same as PostionCrop but grows in size rather than Y Pos.

**Future Ideas:**
I would also like to change the gathering action to get away from the Arma 3 "Stand still and press the same button over and over to get items"
![cropGrow](https://user-images.githubusercontent.com/45827986/171271524-3b564753-df87-4e36-b2af-e6799d9349c7.gif)

Any ideas, suggestions or improvements?

PR https://github.com/EveronLife/EveronLife/pull/66

**Comment by hiimk3v** [comment]
Buyable greenhouses or garden plots you can put in the backyard of a house you own.

**Comment by Arkensor** [comment]
Related: #7
